src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1537,6 +1537,79 @@ bool ONNXConcatOp::inferShapes() {
   return true;
 }
 
+//===----------------------------------------------------------------------===//
+// Split
+
+bool ONNXSplitOp::inferShapes() {
+  if (!getOperand().getType().cast<RankedTensorType>()) {
+    emitError("Input tensor not ranked");
+    return false;
+  }
+
+  int numOfResults = getNumResults();
+  auto inputType = getOperand().getType().cast<RankedTensorType>();
+  auto inputShape = inputType.getShape();
+  auto inputRank = inputShape.size();
+
+  // Checking value of axis parameter.
+  auto axisIndex = axis().getSExtValue();
```

Comment:
Might be a good idea to check axisIndex is in the range ` [-rank, rank-1]` to begin with. If axisIndex is, let's say `2 * (-rank)`, then bumping it up by `rank` won't make it acceptable.

Reply:
Done.